CMSgov/onemac
# OY2-13640, OY2-13093: add date range picker to filter view

Stories:
- https://qmacbis.atlassian.net/browse/OY2-13640
- https://qmacbis.atlassian.net/browse/OY2-13093

Endpoint: https://dm7v0q5ih475s.cloudfront.net/

### Details

Implement date filters in Package view.

### Changes

- Enable filtering on date columns
- Implement their filters as [React Suite DateRangePicker](https://rsuitejs.com/components/date-range-picker) inputs

### Implementation Notes

- Had to refactor filter rendering to support multiple types of filter. This will be helpful for the final filter story as well.

### Test Plan

1. Log in as `helpdeskactive` and go to Package dash
2. Open the Filter sidebar and observe the three new accordions. Smoke test their functionality to ensure they perform intuitively. For the Date Submitted filter at least, verify that changes made in the picker are reflected in the table after clicking OK.
